app.py

```diff
@@ -35,91 +36,105 @@
+                sanitized_title = Utils.sanitize_filename(base_title)
+
+            ydl_opts = {
+                'format': 'bestvideo+bestaudio/best',
+                'outtmpl': os.path.join(app.config['UPLOAD_FOLDER'], sanitized_title + '.%(ext)s'),
+                'noplaylist': True,
+                'keepvideo': True,  # Keep source files for merging
+                'n_threads': 6
+            }
+
+            with yt_dlp.YoutubeDL(ydl_opts) as ydl:
+                result = ydl.extract_info(url, download=True)
+
+                # Post-download file-ext gymnastics
+                if 'requested_formats' in result:
+                    # If separate video and audio were downloaded and merged
+                    merged_ext = result['ext']
+                else:
+                    # If a single file was downloaded; fallback to mp4 (which probably isn't the best idea)
+                    merged_ext = result.get('ext', 'mp4')
+
+            # Return the sanitized file path for processing
+            video_file = os.path.join(app.config['UPLOAD_FOLDER'], sanitized_title + "." + merged_ext)
+            return os.path.abspath(video_file)
+
+        except Exception as e:
+            logging.error(f"Error downloading video: {e}")
+            return None
+
+    @staticmethod
+    def process_media_with_cpp(video_path):
+        try:
+            logging.info(f"Processing video at path: {video_path}")
 
-    except Exception as e:
-        logging.error(f"Error downloading video: {e}")
-        return None
+            result = subprocess.run([
+                './MediaProcessor/build/MediaProcessor', video_path
+            ], capture_output=True, text=True)
 
-def process_video_with_cpp(video_path):
-    try:
-        logging.info(f"Processing video at path: {video_path}")
+            if result.returncode != 0:
+                logging.error(f"Error processing video: {result.stderr}")
+                return None
 
-        result = subprocess.run([
-            './MediaProcessor/build/MediaProcessor', video_path
-        ], capture_output=True, text=True)
+            for line in result.stdout.splitlines():
+                if "Video processed successfully" in line:
+                    processed_video_path = line.split(": ")[1].strip()
+                    return processed_video_path
 
-        if result.returncode != 0:
-            logging.error(f"Error processing video: {result.stderr}")
+            return None
+        except Exception as e:
+            logging.error(f"Error running C++ binary: {e}")
             return None
 
-        for line in result.stdout.splitlines():
-            if "Video processed successfully" in line:
-                processed_video_path = line.split(": ")[1].strip()
-                return processed_video_path
-
-        return None
-    except Exception as e:
-        logging.error(f"Error running C++ binary: {e}")
-        return None
 
 @app.route('/', methods=['GET', 'POST'])
 def index():
```

Comment:
Could you also add a generic URL validation? `urlparse` from `urllib.parse` should be a good start for us; better than nothing!